### ISSUE TITLE
--format trims one char from {{.Names}} each time it is used

### ISSUE THREAD

**Issue by danielhoherd** [docker/docker#21772]
**Output of `docker version`:**

```
$ docker version
Client:
 Version:      1.10.3
 API version:  1.22
 Go version:   go1.6
 Git commit:   20f81dd
 Built:        Mon, 04 Apr 2016 16:27:28 +1200
 OS/Arch:      linux/amd64

Server:
 Version:      1.10.3
 API version:  1.22
 Go version:   go1.6
 Git commit:   20f81dd
 Built:        Mon, 04 Apr 2016 16:27:28 +1200
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
Containers: 7
 Running: 1
 Paused: 0
 Stopped: 6
Images: 91
Server Version: 1.10.3
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 88
 Dirperm1 Supported: true
Execution Driver: native-0.2
Logging Driver: json-file
Plugins:
 Volume: local
 Network: null host bridge
Kernel Version: 4.4.0-16-generic
Operating System: Ubuntu 16.04
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 15.65 GiB
Name: plata
ID: OERF:H3PM:YRZA:W34S:LM4N:QN2S:AYF4:QYYG:DFZV:JPKW:WROU:6LBO
WARNING: No swap limit support
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

Physical machine, but this issue has happened on several other machines I have tested on running on different OSes and different Docker versions.

**Steps to reproduce the issue:**
1.  Run this script:
```
docker ps --format="
1  Names:  {{.Names}}
2  Names:  {{.Names}}
3  Names:  {{.Names}}
4  Names:  {{.Names}}
5  Names:  {{.Names}}
6  Names:  {{.Names}}
7  Names:  {{.Names}}
8  Names:  {{.Names}}
"
```

**Describe the results you received:**
Output of the above script on a machine with one container:
```
1  Names:  bitlbee
2  Names:  itlbee
3  Names:  tlbee
4  Names:  lbee
5  Names:  bee
6  Names:  ee
7  Names:  e
8  Names:
```

This behavior does not occur with the other available format vars.

**Comment by duglin** [comment]
that's pretty funny  :-)

**Comment by thaJeztah** [comment]
Still able to reproduce this on 1.11.0-rc3

**Comment by thaJeztah** [comment]
Could possibly be this; https://github.com/docker/docker/blob/3d13fddd2bc4d679f0eaa68b0be877e5a816ad53/api/client/formatter/custom.go#L231-L237

**Comment by vieux** [comment]
@thaJeztah yes that's it

**Comment by thaJeztah** [comment]
let me have a quick look if I can fix it

**Comment by vieux** [comment]
@thaJeztah I just opened a possible fix in #21783

**Comment by vieux** [comment]
Closed by #21783
